### PR TITLE
Deprecate getters/setters replaced with properties in compile tasks

### PR DESCRIPTION
**Pull request by rieske**
https://github.com/gradle/gradle/issues/15681

- SourceDirectorySet.(g|s)etOutputDir() replaced with destinationDirectory property
- AbstractCompile.java.(g|s)etDestinationDir() replaced with destinationDirectory property

### CODE REVIEW

**Comment by rieske** [comment]
`JavaCompile.setDestinationDir(Provider<File>)` is used by Android plugin:
```
at com.android.build.gradle.tasks.JavaCompileCreationAction.configure(JavaCompile.kt:155)
at com.android.build.gradle.tasks.JavaCompileCreationAction.configure(JavaCompile.kt:49)
```

**Comment by rieske** [comment]
`AbstractCompile.getDestinationDir()` is used by the Kotlin plugin:

```
at org.jetbrains.kotlin.gradle.plugin.KotlinPluginKt$configureJavaTask$1.execute(KotlinPlugin.kt:1000)
at org.jetbrains.kotlin.gradle.plugin.KotlinPluginKt$configureJavaTask$1.execute(KotlinPlugin.kt:998)
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -153,3 +153,23 @@ The last set of classes have no external or internal usages and therefore were d
 
 
 
+
+[[compile_task_wiring]]
+=== Properties deprecated in compile task wiring
```
Comment (donat) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:158]:
```suggestion
=== Deprecated properties in `compile` task
```

File: subprojects/language-jvm/src/main/java/org/gradle/api/tasks/compile/AbstractCompile.java [hunk 2]
```diff
@@ -78,18 +79,37 @@ public DirectoryProperty getDestinationDirectory() {
      * Returns the directory to generate the {@code .class} files into.
      *
      * @return The destination directory.
+     *
+     * @deprecated Use {@link #getDestinationDirectory()} instead. This method will be removed in Gradle 8.0.
      */
     @ReplacedBy("destinationDirectory")
+    @Deprecated
     public File getDestinationDir() {
+        // Used in Kotlin plugin - needs updating there and bumping the version first. Followup with https://github.com/gradle/gradle/issues/16783
```
Comment (donat) [inline subprojects/language-jvm/src/main/java/org/gradle/api/tasks/compile/AbstractCompile.java:88]: 👍

**Review by donat — APPROVED** [review]

**Pull request merged**